doc/index.rst

```diff
@@ -154,6 +154,9 @@ Note that some of these currently tied only to a single option, but we still lea
 * `CATKIN_PARALLEL_JOBS` (default: -p4): Maximum number of packages to be built in parallel that is passed to underlining build tool. As of Jan 2016, this is only enabled with `catkin_tools`. See for more detail about `number of build jobs <http://catkin-tools.readthedocs.org/en/latest/verbs/catkin_build.html#controlling-the-number-of-build-jobs>`_ and `documentation of catkin_tools <https://catkin-tools.readthedocs.org/en/latest/verbs/catkin_build.html#full-command-line-interface>`_ that this env variable is passed to internally in `catkin-tools`.
 * `CATKIN_PARALLEL_TEST_JOBS` (default: -p4): Maximum number of packages which could be examined in parallel during the test run. If not set it's filled by `ROS_PARALLEL_JOBS`.
 * `CI_PARENT_DIR` (default: .ci_config): (NOT recommended to specify) This is the folder name that is used in downstream repositories in order to point to this repo.
+* `DOCKER_IMAGE` (default: not set): Selects a Docker images different from default one. Please note, this disables the handling of `ROS_REPOSITORY_PATH` and `ROS_DISTRO`.
```

Comment:
The idea is no specify the image that corresponds to `ROS_REPOSITORY_PATH` and `ROS_DISTRO`.  ROS needs to be already installed in the image, so it cannot so specified externally. `ROS_DISTRO` is still needed or has to be set in the image.

So instead of `ROS_DISTRO="indigo" ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu`, you can set `DOCKER_IMAGE=myrepo:indigo` or `DOCKER_IMAGE=myrepo:indigo-shadow-fixed`.

If we want to, we can event implement a template mechanism.

> Maybe we want to add a sample section for this and the subsequent 2 new DOCKER_* variables

I'd like to have a chapter (in a splitted documentation) on it, as it is an expert option. The images have to meet some requirements as well.
However, this needs to come later. It has to wait until the automated Docker builds have set-up and we have split the documentation to reflect the new modularity.